Property values can behave unexpectedly with trailing spaces

### Describe the bug

If a trailing space (or presumably another whitespace character) is present at the end of a value in the src/main/resources/application.properties file, it is passed verbatim (not trimmed) to the user of the property (in at least one case, see reproducer below).



### Expected behavior

I would expect trailing (or proceeding) whitespace in a property value to be trimmed. I think it should only be preserved if the value is quoted somehow.

### Actual behavior

In some cases the trailing whitespace remains when the property is used. See the reproducer for one such case.

### How to Reproduce?

Reproducer:

1. Run the https://quarkus.io/guides/opentelemetry example.
2. When populating the `application.properties` file, copy and past from the code block in the guide
3. Observe that. whitespace character has been added to the end of every property value in the `application.properties` file.
4. When starting Quarkus, observe: `Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: http://localhost:4317 `
5.   Remove the traiing white space from `quarkus.opentelemetry.tracer.exporter.otlp.endpoint=http://localhost:4317 ` in the `application.properties`.
6. Re-run Quarkus and observe that the above exception is not thrown.

### Output of `uname -a` or `ver`

Darwin Pauls-Mac-mini-5.local 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.7" 2020-04-14

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) Maven home: /Users/paul/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4 Java version: 11.0.7, vendor: AdoptOpenJDK, runtime: /Users/paul/.sdkman/candidates/java/11.0.7.hs-adpt Default locale: en_GB, platform encoding: UTF-8 OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"

### Additional information

_No response_

--- a/extensions/opentelemetry/opentelemetry-exporter-jaeger/runtime/src/main/java/io/quarkus/opentelemetry/exporter/jaeger/runtime/JaegerExporterConfig.java
+++ b/extensions/opentelemetry/opentelemetry-exporter-jaeger/runtime/src/main/java/io/quarkus/opentelemetry/exporter/jaeger/runtime/JaegerExporterConfig.java
@@ -6,6 +6,8 @@ import java.util.Optional;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.runtime.annotations.ConvertWith;
+import io.quarkus.runtime.configuration.TrimmedStringConverter;
 
 public class JaegerExporterConfig {
     @ConfigRoot(name = "opentelemetry.tracer.exporter.jaeger", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
@@ -25,6 +27,7 @@ public class JaegerExporterConfig {
          * The Jaeger endpoint to connect to. The endpoint must start with either http:// or https://.
          */
         @ConfigItem
+        @ConvertWith(TrimmedStringConverter.class)
         public Optional<String> endpoint;
 
         /**

--- a/extensions/opentelemetry/opentelemetry-exporter-otlp/deployment/src/test/java/io/quarkus/opentelemetry/exporter/otlp/deployment/OtlpExporterConfigTest.java
+++ b/extensions/opentelemetry/opentelemetry-exporter-otlp/deployment/src/test/java/io/quarkus/opentelemetry/exporter/otlp/deployment/OtlpExporterConfigTest.java
@@ -1,0 +1,28 @@
+package io.quarkus.opentelemetry.exporter.otlp.deployment;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.opentelemetry.exporter.otlp.runtime.OtlpExporterConfig.OtlpExporterRuntimeConfig;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class OtlpExporterConfigTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .withEmptyApplication()
+            .overrideConfigKey("quarkus.opentelemetry.tracer.exporter.otlp.endpoint", "http://localhost ");
+
+    @Inject
+    OtlpExporterRuntimeConfig config;
+
+    @Test
+    void config() {
+        assertTrue(config.endpoint.isPresent());
+        assertEquals("http://localhost", config.endpoint.get());
+    }
+}

--- a/extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpExporterConfig.java
+++ b/extensions/opentelemetry/opentelemetry-exporter-otlp/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/runtime/OtlpExporterConfig.java
@@ -7,6 +7,8 @@ import java.util.Optional;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.runtime.annotations.ConvertWith;
+import io.quarkus.runtime.configuration.TrimmedStringConverter;
 
 public class OtlpExporterConfig {
     @ConfigRoot(name = "opentelemetry.tracer.exporter.otlp", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
@@ -25,7 +27,8 @@ public class OtlpExporterConfig {
         /**
          * The OTLP endpoint to connect to. The endpoint must start with either http:// or https://.
          */
-        @ConfigItem()
+        @ConfigItem
+        @ConvertWith(TrimmedStringConverter.class)
         public Optional<String> endpoint;
 
         /**
@@ -36,7 +39,7 @@ public class OtlpExporterConfig {
          * "https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables">
          * Specifying headers</a> for more details.
          */
-        @ConfigItem()
+        @ConfigItem
         Optional<List<String>> headers;
 
         /**
@@ -55,7 +58,7 @@ public class OtlpExporterConfig {
          * "https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#opentelemetry-protocol-exporter">
          * Configuration Options</a> for the supported compression methods.
          */
-        @ConfigItem()
+        @ConfigItem
         public Optional<String> compression;
 
     }

